src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -329,14 +335,20 @@ async def call_api_partial(page):
             self.coroutines_to_execute[operation_id].append(coroutine)
             set_index += 1
 
+
         # Return a dummy request for the SDK to use
         # This allows us to skip right to the AfterRequestHook and await all the calls
+        # Also, pass the operation_id so after_success can await the right results
+
         # Note: We need access to the async_client from the sdk_init hook in order to set
         # up a mock request like this.
         # For now, just make an extra request against our api, which should return 200.
         # dummy_request = httpx.Request("GET",  "http://no-op")
-
-        dummy_request = httpx.Request("GET",  "https://api.unstructuredapp.io/general/docs")
+        dummy_request = httpx.Request(
+            "GET",
+            "https://api.unstructuredapp.io/general/docs",
+            headers={"request_id": operation_id},
```

Comment:
should we call it `operation_id` in the header key as well to avoid confusion?

Reply:
Good call!